src/tokenizer.js

```diff
@@ -258,8 +258,10 @@ export default class Tokenizer {
   }
 
   createError(message) {
-    /* istanbul ignore next */
-    let msg = message.replace(/\{(\d+)\}/g, (_, n) => JSON.stringify(arguments[+n + 1]));
+    let msg = message;
+    if (typeof msg === 'function') {
+      msg = msg.apply(this, [].slice.call(arguments, 1).map(JSON.stringify));
```

Comment:
It would be like

```js
createError(message, ...messageParams) {
  let msg;
  if (typeof message === 'function') {
    msg = message(...messageParams);
  } else {
    msg = message;
  }
  // ...
}
```